app/models/miq_expression.rb

```diff
@@ -1496,7 +1496,12 @@ def self.build_relats(model, parent = {}, seen = [])
     result = {:columns => model.column_names_with_virtual, :parent => parent}
     result[:reflections] = {}
 
-    model.reflections_with_virtual.each do |assoc, ref|
+    refs = model.reflections_with_virtual
+    model.descendants.each do |desc|
+      refs.reverse_merge!(desc.reflections_with_virtual)
+    end if model.respond_to?(:include_descendant_classes_in_expressions?) && model.include_descendant_classes_in_expressions?
```

Comment:
You can use `try` here.  [try will also return nil if the receiver does not respond to the method](http://apidock.com/rails/Object/try).  I also prefer seeing the `if` condition upfront, but I'll leave that up to you.

```ruby
if model.try(:include_descendant_classes_in_expressions?)
  model.descendants.each { |desc| refs.reverse_merge!(desc.reflections_with_virtual) }
end
```